v-next/core/src/internal/global-parameters.ts

```diff
@@ -104,13 +114,43 @@ export function buildGlobalParameterDefinition<T extends ParameterType>({
   };
 }
 
+/**
+ * Resolves the global arguments by parsing the user provided arguments and
+ * environment variables. The arguments are validated against the global
+ * parameter definitions, and the default values are used when the arguments
+ * are not provided. Only the arguments defined in the global parameters map
+ * are resolved.
+ *
+ * @param userProvidedGlobalArguments The arguments provided by the user. These
+ * take precedence over environment variables.
+ * @param globalParametersMap The map of global parameter definitions to
+ * validate the arguments.
+ */
 export function resolveGlobalArguments(
   userProvidedGlobalArguments: Partial<GlobalArguments>,
```

Comment:
I believe the type for `userProvidedGlobalArguments` is incorrect, as the arg values are always strings and might differ from what the user specifies in the type. I suggest using `Record<string, string | undefined>` instead. wdyt, @alcuadrado?